spark/src/main/scala/org/apache/mahout/sparkbindings/blas/ABt.scala

```diff
@@ -116,7 +118,12 @@ object ABt {
           // Empty combiner += value
           createCombiner = (t: (Array[K], Array[Int], Matrix)) =>  {
             val (rowKeys, colKeys, block) = t
-            val comb = new SparseMatrix(prodNCol, block.nrow).t
+
+            val comb = if (block.getFlavor == MatrixFlavor.SPARSELIKE) {
+              new SparseMatrix(prodNCol, block.nrow).t
+            } else {
+              new DenseMatrix(prodNCol, block.nrow).t
+            }
```

Comment:
@dlyubimov I think that the only thing left to do is to use `sparsityAnalysis()` here if it is needed rather than check the matrix structure.